BootloaderCorePkg/Tools/SingleSign.py

```diff
@@ -89,6 +118,14 @@ def single_sign_file (priv_key, hash_type, sign_scheme, in_file, out_file):
         "RSA_PSS"      : 'pss',
     }
 
+    # Check for Key Id or key path
+    if not os.path.exists(priv_key):

```

Comment:
I think the logic should be something like this:

if (os.path.exists(priv_key)):
  use_it

else {
 //Get SLIMBOOT_KEY_DIR path.
//if (priv_key is simple file name) search for file 'priv_key' in SLIMBOOT_KEY_DIR
//if (priv_key is KEY_ID) search for corresponding keyfile  in SLIMBOOT_KEY_DIR
}